src/cloudant/design_document.py

```diff
@@ -149,6 +160,21 @@ def add_search_index(self, index_name, search_func, analyzer=None):
 
         self.indexes.__setitem__(index_name, search)
 
+    def add_list_function(self, list_name, list_func):
+        """
+        Appends a list function to the locally cached DesignDocument
+        indexes dictionary.
+
+        :param list_name: Name used to identify the list function.
+        :param list_func: Javascript list function.
```

Comment:
I think it's better to add `str` (also found in add_search_index):
`:param str list_name: Name used to identify the list function.`
`:param str list_func: Javascript list function.`